packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,633 @@
+
+  constructor(
+    host: ReactiveControllerHost,
+    store: Store<State, Action>,
+    selector: Selector<State, Result>,
+  ) {
+    this._host = host;
+    host.addController(this);
+
+    this._store = store;
+    this._selector = selector;
+  }
+}
+```
+
+Now let’s initialize the initially selected value so that the user can access the state’s initial value with `selectorController.value` using the user's selector:
+
+```ts
+...
+export class SelectorController<
+    State,
+    Action extends RAction,
+    Result = unknown
+> implements ReactiveController {
+  ...
+  value: Result;
+
+  constructor(
+    host: ReactiveControllerHost,
+    store: Store<State, Action>,
+    selector: Selector<State, Result>,
+  ) {
+    ...
+    this.value = selector(store.getState());
+  }
+}
+```
+
+### Updating the Component on Store Update
+
+We now have a controller that initializes to the initial state of the store, next let’s update `this.value` when the state updates and then tell the host element to re-render when we have detected a change in the selected value.
+
+Redux stores have a [`Store.subscribe(listener)`](https://redux.js.org/api/store#subscribelistener) method which will call a given callback whenever the state of the store updates. Let's hook into this, update `this.value`, and tell the host to update when the component is connected to the DOM:
+
+```ts
+...
+export class SelectorController<
+    State,
+    Action extends RAction,
+    Result = unknown
+> implements ReactiveController {
+  ...
+
+  hostConnected() {
+    this.store.subscribe(() => {
+      const selected = this._selector(this._store.getState());
+      if (selected !== this.value) {
+        this.value = selected;
+        this._host.requestUpdate();
+      }
+    });
+  }
+}
+```
+
+Great! Now the controller will update its value and tell the host element to update whenever the state changes. Additionally, by first comparing the previous state to the current state, we can avoid re-rendering components that don't need to be re-rendered which can improve performance. Nice!
+
+In conclusion, we need to improve our component so that it does not re-render when the component is disconnected from the page and the store’s state changes. Redux’s `Store.subscribe()` method returns an `unsubscribe()` function. Let’s keep track of this and unsubscribe from the store’s changes when the component disconnects.
+
+```ts
+...
+export class SelectorController<
+    State,
+    Action extends RAction,
+    Result = unknown
+> implements ReactiveController {
+  ...
+  private _unsubscribe: () => void;
+  ...
+
+  hostConnected() {
+    this._unsubscribe = this._store.subscribe(() => {
+      ...
+    });
+  }
+
+  hostDisconnected() {
+    this._unsubscribe();
+  }
+}
+```
+
+We now have a Redux `SelectorController` Reactive Controller that listens to a Redux store for state changes, selects a value from the state, and updates the host element whenever that state value changes!
```

Comment:
Nice section! Imho this section is the juicy part of the article that shows off the actual integration. We can maybe celebrate this more.
Could use this location to hammer home the benefits of having a `SelectorController`.

"The SelectorController can be reused across components that need to access some state in the redux store, and all the logic for automatically and efficiently re-rendering those components is encapsulated in this controller."